src/www/ui/search.php

```diff
@@ -266,6 +267,20 @@ function GetResults($Item, $Filename, $tag, $Page, $SizeMin, $SizeMax, $searchty
 
     $Offset = $Page * $this->MaxPerPage;
     $SQL .= " ORDER BY ufile_name, uploadtree.pfile_fk";
+    $result = pg_query($PG_CONN, $SQL);
```

Comment:
Since you are only interested in the size of the result of `$SQL` you could tweak the query to directly return the number of rows.

This code also implies that you ask the DB twice, with and without paging. Why then use paging in the first place if one has already fetched all data.

Reply:
This is copied code and should be factored out to a function (in fact there are more elaborate libraries to talk to the DB in the source code)

Reply:
Ok. Thanks for the feedback. I will update the patch soon.

Reply:
Can you please review the approach. 
How about fetching all results without paging in $totalUploadTreeRecs and then just return the $MaxPerPage results according to the page number as $uploadTreeRecs.
``` 
return array($uploadTreeRecs, $totalUploadTreeRecsCount)
``` 
In this way, we will be executing the query only once and no copied code will be there.